src/pages/invoices/common/hooks/useResolveInputField.tsx

```diff
@@ -73,11 +73,14 @@ export function useResolveInputField() {
       return (
         <InputField
           id={property}
-          type="number"
           value={invoice?.line_items[index][property]}
-          onChange={(event: ChangeEvent<HTMLInputElement>) =>
-            onChange(property, parseFloat(event.target.value), index)
-          }
+          onChange={(event: ChangeEvent<HTMLInputElement>) => {
+            if (isNaN(Number(event.target.value)) || event.target.value == '') {
+              event.target.value = '0';
+              return event;
+            }
+            return onChange(property, parseFloat(event.target.value), index);
```

Comment:
This is not good and not convention. Please avoid placing inline logic whenever possible. The extracted method would be much nicer.

Regarding this example, you'll need to extract it into method anyways, because we want to perform this validation for the recurring invoices as well.